lms/djangoapps/onboarding_survey/history.py

```diff
@@ -97,27 +120,52 @@ def add_organization_survey(history, user):
         history(History): History table model
         user(User): Django user model instance
     """
+    extended_profile = user.extended_profile
+    if extended_profile.is_poc or is_first_signup_in_org(extended_profile.organization):
+        # Partner Network
+        many_to_many_fields_mapping = {
+            'Mercy Corps': 'partner_network_mercy_corps',
+            'Global Giving': 'partner_network_global_giving',
+            'FHI 360 / FHI Foundation': 'partner_network_fhi_360',
+            '+Acumen': 'partner_network_acumen',
+        }
+        # Some of the fields in history table have different corresponding names in organization survey.
+        history_and_org_survey_field_mappings = {
+            'org_sector': 'sector',
+            'org_level_of_operation': 'level_of_operation',
+            'org_focus_area': 'focus_area',
+            'org_total_employees': 'total_employees',
+            'org_country': 'country',
+            'org_city': 'city',
+            'org_is_url_exist': 'is_org_url_exist',
+            'org_url': 'url',
+            'org_founding_year': 'founding_year',
+            'org_alternate_admin_email': 'alternate_admin_email'
+        }
+
+        try:
+            organization_survey = user.organization_survey
+            add_fields([], organization_survey, history, history_and_org_survey_field_mappings)
+
+            for partner_network in organization_survey.partner_network.all():
+                setattr(history, many_to_many_fields_mapping[partner_network.name], True)
+        except OrganizationSurvey.DoesNotExist:
+            pass
```

Comment:
Could you please put some log in case of exception?

Reply:
Done